packages/core/src/lib/light_push/light_push.ts

```diff
@@ -135,27 +209,95 @@ export class LightPushCore {
       bytes.append(chunk);
     });
 
+    if (bytes.length === 0) {
+      return {
+        success: null,
+        failure: {
+          error: LightPushError.NO_RESPONSE,
+          peerId: peerId
+        }
+      };
+    }
+
+    // Handle response based on protocol version
+    if (protocol === LightPushCodecV3) {
+      return this.handleV3Response(bytes, peerId);
+    } else {
+      return this.handleV2Response(bytes, peerId);
+    }
+  }
```

Comment:
I think it's risky to have to remember to do different protocol checks. I would instead have `send` delegate to `sendV2` or `sendV3` once the protocol is checked once.